R/doeAnalysis.R

```diff
@@ -528,7 +519,28 @@ doeAnalysis <- function(jaspResults, dataset, options, ...) {
     ggplot2::scale_y_continuous(name = gettext("Residuals"), breaks = yBreaks, limits = range(yBreaks)) +
     jaspGraphs::geom_rangeframe() +
     jaspGraphs::themeJaspRaw()
-  plot$plotObject <- p
+  return(p)
+}
+
+.doeAnalysisPlotMatrixResidualPlot <- function(jaspResults, dataset, options, ready) {
+  if (!is.null(jaspResults[["fourInOneResidualPlot"]]) || !options[["fourInOneResidualPlot"]]) {
+    return()
+  }
+  plot <- createJaspPlot(title = gettext("Matrix residual plot"), width = 1000, height = 1000)
+  plot$dependOn(options = c("fourInOneResidualPlot", .doeAnalysisBaseDependencies()))
+  plot$position <- 11
+  jaspResults[["fourInOneResidualPlot"]] <- plot
+  if (!ready || is.null(jaspResults[["doeResult"]]) || jaspResults$getError()) {
+    return()
+  }
+  result <- jaspResults[["doeResult"]]$object[["regression"]]
+  plotList <- list()
+  plotList[[1]] <- .doeAnalysisPlotResidualsVsOrderPlotObject(jaspResults, dataset, options)
+  plotList[[2]] <- .doeAnalysisPlotFittedVsResidualsPlotObject(jaspResults, options)
+  plotList[[3]] <- jaspDescriptives::.plotMarginal(resid(result[["object"]]), NULL)
+  plotList[[4]] <- jaspGraphs::plotQQnorm(resid(result[["object"]]))
+  plotMat <- matrix(plotList, 2, 2)
```

Comment:
how about making the matrix directly?
```suggestion
  plotMat <- matrix(list(), 2, 2)
  plotMat[[1, 1]] <- .doeAnalysisPlotResidualsVsOrderPlotObject(jaspResults, dataset, options)
  plotMat[[2, 1]] <- .doeAnalysisPlotFittedVsResidualsPlotObject(jaspResults, options)
  plotMat[[1, 2]] <- jaspDescriptives::.plotMarginal(resid(result[["object"]]), NULL)
  plotMat[[2, 2]] <- jaspGraphs::plotQQnorm(resid(result[["object"]]))
```